nixos/modules/security/wrappers/wrapper.c

```diff
@@ -16,10 +16,7 @@
 #include <syscall.h>
 #include <byteswap.h>
 
-// Make sure assertions are not compiled out, we use them to codify
-// invariants about this program and we want it to fail fast and
-// loudly if they are violated.
-#undef NDEBUG
+#define ASSERT(expr) (expr) ? (void) 0 : assert_failure(#expr)
```

Comment:
Question on hygiene here: Wouldn't the result of this define need an additional pair of parentheses to avoid nasty C bugs like:

```c
if (something || ASSERT(1 == 1))
// if (something || (1 == 1) ? (void) 0 : assert_failure("1 == 1"))
```

which would be equal to `(something || 1 == 1) ? void (0) : assert_failure("1 == 1)`?

(I'm not into C for a few years,... but that catched my eye.

Reply:
That's a fair point, fixed.